internal/services/controller/controller.go

```diff
@@ -259,19 +272,49 @@ func (c *Controller) Run(ctx context.Context) error {
 }
 
 func (c *Controller) startConditionalInformersWithWatcher(ctx context.Context, conditionalInformers []conditionalInformer) {
+	tryConditionalInformers := conditionalInformers
 	if err := wait.PollImmediateInfiniteWithContext(ctx, time.Minute*2, func(ctx context.Context) (done bool, err error) {
-		apiResourceLists := fetchApiResourceLists(c.discovery, c.log)
+		apiResourceLists := fetchAPIResourceLists(c.discovery, c.log)
 		if apiResourceLists == nil {
 			return false, nil
 		}
+		c.log.Infof("Cluster API server is available, trying to start conditional informers")
 
-		c.log.Infof("Cluster API server is available, starting conditional informers")
-		for _, informer := range conditionalInformers {
-			apiResourceListForGroupVersion := getApiResourceListByGroupVersion(informer.groupVersion, apiResourceLists)
-			if isResourceAvailable(informer.apiType, apiResourceListForGroupVersion) {
-				custominformers.NewHandledInformer(c.log, c.queue, informer.informer, informer.apiType, nil)
+		for _, informer := range tryConditionalInformers {
+			if informer.isApplied {
+				continue
 			}
+			apiResourceListForGroupVersion := getAPIResourceListByGroupVersion(informer.groupVersion, apiResourceLists)
+
+			resourceAvailable := isResourceAvailable(informer.apiType, apiResourceListForGroupVersion)
+			informerHasAccess := c.informerHasAccess(ctx, informer)
+
+			if resourceAvailable && informerHasAccess {
+				c.log.Infof("Starting conditional informer for %v", informer.name)
+
+				informer.informer = applyInformer(informer.groupVersion, c.informerFactory)
+				handledInformer := custominformers.NewHandledInformer(c.log, c.queue, informer.informer, informer.apiType, nil)
+
+				informer.isApplied = true
+
+				go handledInformer.Run(ctx.Done())
+			} else {
+				c.log.Warnf("Skipping conditional informer name: %v, API resource available: %t, has required access: %t",
+					informer.name,
+					resourceAvailable,
+					informerHasAccess,
+				)
+			}
+		}
+
+		filterNotAppliedConditionInformers := lo.Filter(tryConditionalInformers, func(informer conditionalInformer, _ int) bool {
+			return !informer.isApplied
+		})
+		if len(filterNotAppliedConditionInformers) > 0 {
+			return false, nil
 		}
+
+		<-ctx.Done()
```

Comment:
Is there any reason we retry the function when all informers were applied? The goal should be to only retry the informers that weren't applied. If all the informers were applied, the poller function can exit: return true.